File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/internal/utils/BlockchainEventsCreator.java
package kin.sdk.internal.utils;


import kin.base.Server;
import kin.sdk.internal.services.BlockchainEvents;
import kin.sdk.internal.services.BlockchainEventsImpl;

public class BlockchainEventsCreator {

    private final Server server;

    public BlockchainEventsCreator(Server server) {
        this.server = server;
    }

    public BlockchainEvents create(String accountId) {
        return new BlockchainEventsImpl(server, accountId);
    }
}

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/internal/utils/BlockchainEventsCreator.kt
package kin.sdk.internal.utils

import kin.base.Server
import kin.sdk.internal.services.BlockchainEvents
import kin.sdk.internal.services.BlockchainEventsImpl

interface BlockchainEventsCreator {
    fun create(accountId: String): BlockchainEvents
}

internal class BlockchainEventsCreatorImpl(private val server: Server) : BlockchainEventsCreator {
    override fun create(accountId: String): BlockchainEvents {
        return BlockchainEventsImpl(server, accountId)
    }
}

